### ISSUE TITLE
Default Pseudo-Versioning Method switches back to APK Link Hash

### ISSUE THREAD

**Issue by 5FeetUnder** [ImranR98/Obtainium#1373]
**Describe the bug**

I changed the "Default Pseudo-Versioning Method" for an app to "Partial APK Hash" and updated the app. After a while the Method is back to "APK Link Hash" for that same app.

**To Reproduce**

I wanted to try out that Pseudo-Versioning method for an app where the APK link is always the same without any version information (something like "website.of.dev/download/app.apk")

**Comment by ImranR98** [comment]
Will you provide the app URL?

**Comment by 5FeetUnder** [comment]
Of course. It happens for me with both proton calendar and proton drive:

https://proton.me/download/CalendarAndroid/ProtonCalendar-Android.apk

https://proton.me/download/DriveAndroid/ProtonDrive-Android.apk

**Comment by ImranR98** [comment]
Here's what I think is happening. The APK link hash doesn't actually hash the entire APK because then you'd have to download the entire APK and that's not practical just for an update check. So instead we try to hash the first few kilobytes and we repeat this process a few times to make sure the hash is consistent. The reason I had to add this repetition is because for some reason that I can't figure out, some sources send a slightly different APK each time (maybe some file metadata changes). When this happens, Obtainium decides that APK hashing is not reliable and falls back to hashing the link.

I haven't actually checked the proton APK to confirm that this is what's going on but if it is then it could probably be communicated in a better way so I'll add a UI element somewhere for that.

**Comment by vvirtues** [comment]
Might be unrelated, but would I use HTML source or Direct APK link for these apps? Using HTML seems to crash my app but direct link is fine

**Comment by ImranR98** [comment]
> Here's what I think is happening...

Got around to actually checking the code, and what I wrote was incorrect. Obtainium does automatically adjust some settings but the "pseudo-versioning method" is not one of them. If partial APK hashing fails, it just errors out and expects the user to change the setting manually.

So I'm not sure what caused the behaviour reported by @5FeetUnder. Strangely it doesn't happen for me, at least right now.
